src/ovs/ovs_control.cpp

```diff
@@ -100,654 +100,615 @@ bool OVS_Control::bundle;
+  if (usable_protocols & allowed_protocols) {
+    // ovs_fatal(0, "switch does not support any of the usable flow "
+    //           "formats (%s)", usable_s);
+    ACA_LOG_ERROR("switch does not support any of the usable flow "
+                  "formats (%s)",
+                  usable_s);
+  } else {
+    char *allowed_s = ofputil_protocols_to_string(allowed_protocols);
+    // ovs_fatal(0, "none of the usable flow formats (%s) is among the "
+    //           "allowed flow formats (%s)", usable_s, allowed_s);
+    ACA_LOG_ERROR("none of the usable flow formats (%s) is among the "
+                  "allowed flow formats (%s)",
+                  usable_s, allowed_s);
+  }
+  return (ofputil_protocol)0;
 }
 
-int
-OVS_Control::add_flow(const char *bridge, const char *flow)
+int OVS_Control::add_flow(const char *bridge, const char *flow)
 {
-    return flow_mod(bridge, flow, OFPFC_ADD);
+  return flow_mod(bridge, flow, OFPFC_ADD);
 }
 
-int
-OVS_Control::mod_flows(const char *bridge, const char *flow, bool strict)
+int OVS_Control::mod_flows(const char *bridge, const char *flow, bool strict)
 {
-    return flow_mod(bridge, flow, strict ? OFPFC_MODIFY_STRICT : OFPFC_MODIFY);
+  return flow_mod(bridge, flow, strict ? OFPFC_MODIFY_STRICT : OFPFC_MODIFY);
 }
 
-int
-OVS_Control::del_flows(const char *bridge, const char *flow, bool strict)
+int OVS_Control::del_flows(const char *bridge, const char *flow, bool strict)
 {
-    return flow_mod(bridge, flow, strict ? OFPFC_DELETE_STRICT : OFPFC_DELETE);
+  return flow_mod(bridge, flow, strict ? OFPFC_DELETE_STRICT : OFPFC_DELETE);
 }
 
-int
-OVS_Control::flow_mod(const char *bridge, const char *flow, unsigned short int command)
+int OVS_Control::flow_mod(const char *bridge, const char *flow, unsigned short int command)
 {
-    struct ofputil_flow_mod fm;
-    char *error;
-    enum ofputil_protocol usable_protocols;
-    
-    error = parse_ofp_flow_mod_str(&fm, flow,
-                                    ports_to_accept(bridge),
-                                    tables_to_accept(bridge), command,
-                                    &usable_protocols);
-    if (error) {
-        // ovs_fatal(0, "%s", error);
-        ACA_LOG_ERROR("%s", error);
-        return EXIT_FAILURE;
-    }
+  ACA_LOG_DEBUG("%s", "OVS_Control::flow_mod ---> Entering\n");
+
+  struct ofputil_flow_mod fm;
+  char *error;
+  enum ofputil_protocol usable_protocols;
+  int rc;
+
+  ACA_LOG_INFO("Executing flow_mod on bridge: %s, flow: %s, command: %d\n",
+               bridge, flow, command);
+
+  auto openflow_client_start = chrono::steady_clock::now();
+
+  error = parse_ofp_flow_mod_str(&fm, flow, ports_to_accept(bridge),
+                                 tables_to_accept(bridge), command, &usable_protocols);
+  if (error) {
+    // ovs_fatal(0, "%s", error);
+    ACA_LOG_ERROR("%s", error);
+    rc = EXIT_FAILURE;
+  } else {
+    // flow_mod__ returns void
     flow_mod__(bridge, &fm, 1, usable_protocols);
```

Comment:
@cj-chung - we are passing in 1 as the number of flows to flow_mod__ here, therefore, I think it wil require some work to fully light up the "bundle" option.

Reply:
for reference, in ovs code, there is parse_ofp_flow_mod_file which has ability to parse multiple flow messages, but I don't like the approach of writing flow message to a file and then read it for bundling.

Reply:
@er1cthe0ne  If flows as bundle come from a file, we need to add the following code or function:
```
static void
ofctl_flow_mod_file(int argc OVS_UNUSED, char *argv[], int command)
{
    enum ofputil_protocol usable_protocols;
    struct ofputil_flow_mod *fms = NULL;
    size_t n_fms = 0;
    char *error;

    if (command == OFPFC_ADD) {
        /* Allow the file to specify a mix of commands.  If none specified at
         * the beginning of any given line, then the default is OFPFC_ADD, so
         * this is backwards compatible. */
        command = -2;
    }
    error = parse_ofp_flow_mod_file(argv[2], ports_to_accept(argv[1]),
                                    tables_to_accept(argv[1]), command,
                                    &fms, &n_fms, &usable_protocols);
    if (error) {
        ovs_fatal(0, "%s", error);
    }
    ofctl_flow_mod__(argv[1], fms, n_fms, usable_protocols);
    free(fms);
}
```
Or we need to count `n_fms` (the size of fms) from the bundle flows.

Reply:
It's better create another PR for the bundle feature improvement.